src/posts/components/PostsList/components/PostCard.tsx

```diff
@@ -0,0 +1,240 @@
+import { useMutation } from '@tanstack/react-query';
+import { Link } from 'react-router';
+
+import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
+import { useAuth } from '@/contexts/AuthContext';
+import useGetBoardItems from '@/hooks/useGetBoardItems';
+import supabase from '@/lib/supabase';
+import { formatRelativeTime } from '@/lib/utils';
+import { PostWithUser } from '@/posts/types';
+
+export const PostCard = ({
+  id,
+  title,
+  status,
+  user: postUser,
+  votes,
+  board,
+  created_at,
+  is_pinned,
+  comments_count,
+  votes_count,
+  refetch,
+}: PostWithUser & { refetch: VoidFunction }) => {
+  const { boards } = useGetBoardItems();
+  const { user } = useAuth();
+
+  const postBoard = boards.find((b) => b?.id === board);
+
+  const statusStyles: { [k in NonNullable<PostWithUser['status']>]: string } = {
+    in_progress: 'bg-yellow-500/10 text-yellow-500 border-yellow-500/10',
+    planned: 'bg-purple-500/10 text-purple-500 border-purple-500/10',
+    completed: 'bg-green-500/10 text-green-500 border-green-500/10',
+    pending: 'bg-gray-500/10 text-gray-500 border-gray-500/10',
+    rejected: 'bg-red-500/10 text-red-500 border-red-500/10',
+    closed: 'bg-blue-500/10 text-blue-500 border-blue-500/10',
+  };
+
+  const statusText: { [k in NonNullable<PostWithUser['status']>]: string } = {
+    in_progress: 'In Progress',
+    planned: 'Planned',
+    completed: 'Completed',
+    pending: 'Pending',
+    rejected: 'Rejected',
+    closed: 'Closed',
+  };
```

Comment:
these will be reinit in every rendering if it's in a component like this, could we move it outside?